donut/modules/directory_search/templates/search_results.html

```diff
@@ -1,44 +1,106 @@
+		{% endfor %}
+	</select>
+</div>
+
+<table class='table table-striped table-condensed'>
+	<thead>
+		<tr>
+			<th>Name</th>
+			<th>Graduation Year</th>
+			{% if show_images %}
+			<th> Image </th>
+			{% endif %}
+		</tr>
+	</thead>
+	<tbody>
+		{% for user in users %}
+		<tr>
+			<td>
+				<a href='{{ url_for("directory_search.view_user", user_id=user["user_id"]) }}'>
+					{{ user['full_name'] }}
+				</a>
+			</td>
+			<td>{{ user['graduation_year'] or '' }}</td>
+			{% if show_images %}
+			<td>
+				{% if user['image'] %}
+				<img
+				src='{{ url_for("directory_search.get_image", user_id=user["user_id"]) }}'
+				style='max-width: 100px; max-height: 150px'
+				/>
+				{% endif %}
+			</td>
+			{% endif %}
+		</tr>
+		{% endfor %}
+	</tbody>
+</table>
+<ul class="pagination">
+	{% set num_pgs = total // per_page + (total % per_page > 0) %}
+	<li class="page-item {% if page == 1 %} disabled {% endif %}">
+		<a class="page-link" {% if page != 1 %} href="{{ url_for('directory_search.search', page=page-1, per_page=per_page, total=total, **query_info) }}" {% endif %} aria-label="Previous">
+			<span aria-hidden="true">&laquo;</span>
+			<span class="sr-only">Previous</span>
+		</a>
+	</li>
+	{% if num_pgs > 15 %}
+	{% if page - 6 > 1 %}
+	<li class="page-item">
+		<a class="page-link" href="{{ url_for('directory_search.search', page=1, per_page=per_page, total=total, **query_info) }}">1</a>
+	</li>
+	<li class="page-item">
+		<a class="page-link">...</a>
+	</li>
+	{% endif %}
+	{% for i in range([1, page - 6] | max , page + 6) %}
+	<li class="page-item {% if i == page %} active {% endif %}">
+		<a class="page-link" href="{{ url_for('directory_search.search', page=i, per_page=per_page, total=total, **query_info) }}">{{ i }}</a>
+	</li>
+	{% endfor %}
+	{% if page + 6 < num_pgs %}
+	<li class="page-item">
+		<a class="page-link">...</a>
+	</li>
+	<li class="page-item">
+		<a class="page-link" href="{{ url_for('directory_search.search', page=num_pgs, per_page=per_page, total=total, **query_info) }}">{{ num_pgs }}</a>
+	</li>
+	{% endif %}
+	{% else %}
+	{% for i in range(1, num_pgs+1) %}
+	<li class="page-item {% if i == page %} active {% endif %}">
+		<a class="page-link" href="{{ url_for('directory_search.search', page=i, per_page=per_page, total=total, **query_info) }}">{{ i }}</a>
+	</li>
+	{% endfor %}
+	{% endif %}
+	<li class="page-item {% if page == num_pgs %} disabled {% endif %}">
+		<a class="page-link" {% if page != num_pgs %} href="{{ url_for('directory_search.search', page=page+1, per_page=per_page, total=total, **query_info) }}" {% endif %} aria-label="Next">
+			<span aria-hidden="true">&raquo;</span>
+			<span class="sr-only">Next</span>
+		</a>
+	</li>
+
+</ul>
+{% endif %}
+</div>
+{% endblock %}
+{% block scripts %}
+{{ super() }}
+<script>
+	$(document).ready(function() {
+		$('#per_page').change(function() {
+			var p = $( "#per_page option:selected" ).text(); 
```

Comment:
What's wrong with `$(this).val()`?

Reply:
It turns into the variable `i`.

Reply:
Oh, because you need to use `value={{ i }}` in the Jinja template